src/Algebra/Morphism/Bundles.agda

```diff
@@ -0,0 +1,217 @@
+
+record MagmaHomomorphism (A : Magma a ℓa) (B : Magma b ℓb) : Set (a ⊔ b ⊔ ℓa ⊔ ℓb) where
+  private
+    module A = Magma A
+    module B = Magma B
+
+  field
+    ⟦_⟧ : A.Carrier → B.Carrier
+
+    isMagmaHomomorphism : IsMagmaHomomorphism A.rawMagma B.rawMagma ⟦_⟧
+
+  open IsMagmaHomomorphism isMagmaHomomorphism public
+
+  setoidHomomorphism : SetoidHomomorphism A.setoid B.setoid
+  setoidHomomorphism = record { isRelHomomorphism = isRelHomomorphism }
+
+------------------------------------------------------------------------
+-- Morphisms between Monoids
+------------------------------------------------------------------------
+
+record MonoidHomomorphism (A : Monoid a ℓa) (B : Monoid b ℓb) : Set (a ⊔ b ⊔ ℓa ⊔ ℓb) where
+  private
+    module A = Monoid A
+    module B = Monoid B
+
+  field
+    ⟦_⟧ : A.Carrier → B.Carrier
+
+    isMonoidHomomorphism : IsMonoidHomomorphism A.rawMonoid B.rawMonoid ⟦_⟧
+
+  open IsMonoidHomomorphism isMonoidHomomorphism public
+
+  magmaHomomorphism : MagmaHomomorphism A.magma B.magma
+  magmaHomomorphism = record { isMagmaHomomorphism = isMagmaHomomorphism }
+
+------------------------------------------------------------------------
+-- Morphisms between Groups
+------------------------------------------------------------------------
+
+record GroupHomomorphism (A : Group a ℓa) (B : Group b ℓb) : Set (a ⊔ b ⊔ ℓa ⊔ ℓb) where
+  private
+    module A = Group A
+    module B = Group B
+
+  field
+    ⟦_⟧ : A.Carrier → B.Carrier
+
+    isGroupHomomorphism : IsGroupHomomorphism A.rawGroup B.rawGroup ⟦_⟧
+
+  open IsGroupHomomorphism isGroupHomomorphism public
+
+  monoidHomomorphism : MonoidHomomorphism A.monoid B.monoid
+  monoidHomomorphism = record { isMonoidHomomorphism = isMonoidHomomorphism }
+
+------------------------------------------------------------------------
+-- Morphisms between NearSemirings
+------------------------------------------------------------------------
+
+record NearSemiringHomomorphism (A : NearSemiring a ℓa) (B : NearSemiring b ℓb) : Set (a ⊔ b ⊔ ℓa ⊔ ℓb) where
+  private
+    module A = NearSemiring A
+    module B = NearSemiring B
+
+  field
+    ⟦_⟧ : A.Carrier → B.Carrier
+
+    isNearSemiringHomomorphism : IsNearSemiringHomomorphism A.rawNearSemiring B.rawNearSemiring ⟦_⟧
+
+  open IsNearSemiringHomomorphism isNearSemiringHomomorphism public
+
+  +-monoidHomomorphism : MonoidHomomorphism A.+-monoid B.+-monoid
+  +-monoidHomomorphism = record { isMonoidHomomorphism = +-isMonoidHomomorphism }
+
+  *-magmaHomomorphism : MagmaHomomorphism A.*-magma B.*-magma
+  *-magmaHomomorphism = record { isMagmaHomomorphism = *-isMagmaHomomorphism }
+
+------------------------------------------------------------------------
+-- Morphisms between Semirings
+------------------------------------------------------------------------
+
+record SemiringHomomorphism (A : Semiring a ℓa) (B : Semiring b ℓb) : Set (a ⊔ b ⊔ ℓa ⊔ ℓb) where
```

Comment:
This seems to be missing a definition of `+-monoidHomomorphism` (which should be reexported downstream)

Actually, shouldn't `SemiringHomomorphism`s and `NearSemiringHomomorphism`s relate?

Reply:
Good catch!
The NB above noted that I'm not always good at spotting missing exports...

Reply:
That's what prompted me to look ;)